### PR TITLE
Avatar URLs are no longer stored

### PR DESCRIPTION
**Pull request by np**
* Avatar URLs are computed only from the user id and avatar dimension
* Convenience functions are provided to access different dimensions

### CODE REVIEW

File: src/DemoData.hs [hunk 1]
```diff
@@ -172,20 +162,27 @@ genCommentVote comments_in_context students = do
     let action = addWithUser . AddCommentVote $ comment ^. _Key
     action student <$> arb
 
-updateAvatar :: User -> URL -> forall m . ActionM m => m ()
-updateAvatar user url = update $ SetUserAvatar (user ^. _Id) url
+updateAvatar :: User -> FilePath -> forall m . ActionM m => m ()
+updateAvatar user file = readImageFile file >>= \case
+    Right pic -> saveAvatar (user ^. _Id) pic
+    Left err -> fail err
```
Comment (andorp) [inline src/DemoData.hs]: We should use `addMessage` and not fail here.

Comment (np) [inline src/DemoData.hs]: I don't mind the use of addMessage but so far this is in DemoData.

**Comment by np** [comment]
Notice that this commit also fixes the following issue:

* New user creates contents but has no avatar yet (`metaCreatedByAvatar = Nothing` on the new created objects)
* The user then changes avatar.
* Contents created before that still has no image.

File: src/DemoData.hs (hunk 1 above)
Comment (andorp) [inline src/DemoData.hs]: Sorry, missed the demo data part. In this case I am ok just having the `fail` combinator here.

**Comment by andorp** [comment]
LGTM

**Pull request merged**